daml-foundations/daml-ghc/daml-stdlib-src/DA/Internal/Template.daml

```diff
@@ -85,6 +85,21 @@ exerciseByKey k e = do
     (c, cc) <- fetchByKey @c k
     internalExerciseWithActors (choiceController cc e) c e
 
+-- | Eexercise a choice on a contract given by its key, conditional on it existing.
+--
+-- The `c` needs to be passed using an explicit type application. For
+-- instance, if you want to exercise a choice `Withdraw` on a contract of
+-- template `Account` given by its key `k`, you must call
+-- `exerciseWhenKey @Account k Withdraw`.
+exerciseWhenKey : forall c k e r.(TemplateKey c k, Choice c e r) => k -> e -> Update (Optional r)
```

Comment:
We should bikeshed the name some more...

Reply:
I still like `exerciseIfExists`

Reply:
@neil-da , @hurryabit , would appreciate if we could move this forwards. If this function is too confusing for us to be able to fund a name for it, I'll close the PR and just include it in my projects ad-hoc.

Reply:
I suck at naming things and @hurryabit is on holiday today. Afraid it will need to wait a day or so.